sdk/eventhub/azure-eventhubs/azure/eventhub/_producer_client.py

```diff
@@ -0,0 +1,170 @@
+                self._producers[producer_index]._closed:  # pylint:disable=protected-access
+            with self._producers_locks[producer_index]:
+                if self._producers[producer_index] is None:
+                    self._producers[producer_index] = self._create_producer(partition_id=partition_id)
+
+        with self._producers_locks[producer_index]:
+            self._producers[producer_index].send(event_data, **kwargs)
+
+    def create_batch(self, max_size=None):
+        # type:(int) -> EventDataBatch
+        """
+        Create an EventDataBatch object with max size being max_size.
+        The max_size should be no greater than the max allowed message size defined by the service side.
+
+        :param int max_size: The maximum size of bytes data that an EventDataBatch object can hold.
+        :rtype: ~azure.eventhub.EventDataBatch
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_eventhub.py
+                :start-after: [START eventhub_producer_client_create_batch_sync]
+                :end-before: [END eventhub_producer_client_create_batch_sync]
+                :language: python
+                :dedent: 4
+                :caption: Create EventDataBatch object within limited size
+
+        """
+        # pylint: disable=protected-access
+        if not self._max_message_size_on_link:
+            self._init_locks_for_producers()
+            with self._producers_locks[-1]:
+                if self._producers[-1] is None:
+                    self._producers[-1] = self._create_producer(partition_id=None)
+                    self._producers[-1]._open_with_retry()  # pylint: disable=protected-access
+            with self._client_lock:
+                self._max_message_size_on_link =\
+                    self._producers[-1]._handler.message_handler._link.peer_max_message_size \
+                    or constants.MAX_MESSAGE_LENGTH_BYTES
+
+        if max_size and max_size > self._max_message_size_on_link:
+            raise ValueError('Max message size: {} is too large, acceptable max batch size is: {} bytes.'
+                             .format(max_size, self._max_message_size_on_link))
+
+        return EventDataBatch(max_size=(max_size or self._max_message_size_on_link))
+
+    def close(self):
+        # type: () -> None
+        """
+        Close down the client. If the client has already closed, this will be a no op.
+
+        :rtype: None
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_eventhub.py
+                :start-after: [START eventhub_producer_client_close_sync]
+                :end-before: [END eventhub_producer_client_close_sync]
+                :language: python
+                :dedent: 4
+                :caption: Close down the client.
+
+        """
+        if self._producers:
+            for p in self._producers:
+                if p:
+                    p.close()
```

Comment:
Minor: I would use a full name (e.g. `producer`) rather than the letter `p` here :)